### DISCUSSION TITLE
how to prevent multiple lines/block in my editor

### DISCUSSION THREAD

**Discussion by costaluu** [codex-team/editor.js#2504] [Q&A]
Hello is possible to have a behavior similar to `<input type="text" />` instead of `<textarea/>` ?

**Comment by sjoerd222888** [comment]
I would also be very interested in this. The best I came up with so far is the following:

```
onChange: async (api, event) => {
  if (event.type === "block-added"){
    api.blocks.delete(1);
  }
  //...
}
```

But this is not nice UX as the block appears and gets deleted again afterwards... I would like to prevent this in the first end so you can have only exactly one paragraph.

**Reply by sjoerd222888** [reply]
My next best solution is inspired by [this](https://github.com/codex-team/editor.js/discussions/1886#discussioncomment-5014782) 

Basically the idea is to create a custom paragraph block tool and handle the keydown event in the drawView method:

```
if(!this.config.multiline){
  div.addEventListener('keydown', (e) =>{
    if (e.code === 'Enter' || e.code === "NumpadEnter") {
      e.preventDefault();
      e.stopPropagation();
    }
  });
}
```

Still would be great if prevention of multiple would be possible out of the box.
